### PR TITLE
Add test for SYCL 2020 specialization constants

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

File: SYCL/SpecConstants/2020/handler-api.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+// - test that specialization constant values can be set within command group
+//   scope and correctly retrieved within a kernel
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// FIXME: SYCL 2020 specialization constants are not supported on host device
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// FIXME: ACC devices use emulation path, which is not yet supported
+// FIXME: CUDA uses emulation path, which is not yet supported
+// UNSUPPORTED: cuda
```
Comment (kbobrovs) [inline SYCL/SpecConstants/2020/handler-api.cpp:16]: \+ UNSUPPORTED: acc?

File: SYCL/SpecConstants/2020/handler-api.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+             "integer specialization constant (defined without default value)"))
+    return false;
+
+  auto double_acc = double_buffer.get_access<sycl::access::mode::read>();
+  if (!check_value(3.14, double_acc[0], "double specialization constant"))
+    return false;
+
+  auto custom_type_acc =
+      custom_type_buffer.get_access<sycl::access::mode::read>();
+  const custom_type custom_type_ref;
```
Comment (kbobrovs) [inline SYCL/SpecConstants/2020/handler-api.cpp:99]: Nit: the default value of the `custom_type_id` coincides with default CPP (zeroes). the test could be strengthened by assigning custom_type_id some specific non-zero value to verify new SYCL2020 capabilities work for non-primitive types as well.

File: SYCL/SpecConstants/2020/handler-api.cpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    return false;
+
+  auto custom_type_acc =
+      custom_type_buffer.get_access<sycl::access::mode::read>();
+  if (!check_value(new_custom_type_value, custom_type_acc[0],
+                   "custom_type specialization constant"))
+    return false;
+
+  return true;
+}
```
Comment (kbobrovs) [inline SYCL/SpecConstants/2020/handler-api.cpp:197]: For completeness, kernel_bundle-API based should also be tested. I guess these would be dozen extra lines of code. Can be addressed in a separate PR of course.

**Review by kbobrovs — DISMISSED** [review]

File: SYCL/SpecConstants/2020/handler-api.cpp (hunk 1 above)
Comment (AlexeySachkov) [reply SYCL/SpecConstants/2020/handler-api.cpp:16]: I don't have `ACC_RUN_PLACEHOLDER`, so it is not necessary

File: SYCL/SpecConstants/2020/handler-api.cpp (hunk 2 above)
Comment (AlexeySachkov) [reply SYCL/SpecConstants/2020/handler-api.cpp:99]: > Nit: the default value of the `custom_type_id` coincides with default CPP (zeroes). 

It should have some non-zero default values, as I have its fields initialized: https://github.com/intel/llvm-test-suite/pull/255/files#diff-308baade624f8087060f981e03c24cffa77255b25f54deb6f7b7f1f6349a84afR16

File: SYCL/SpecConstants/2020/handler-api.cpp (hunk 3 above)
Comment (AlexeySachkov) [reply SYCL/SpecConstants/2020/handler-api.cpp:197]: > For completeness, kernel_bundle-API based should also be tested. I guess these would be dozen extra lines of code.

Sure, I definitively don't want to add those into the same test as it is huge enough already, but a separate file with such tests is needed for sure

**Comment by AlexeySachkov** [comment]
@vladimirlaz, if I recall correctly, @kbobrovs approved this PR once. Since then only formatting changed - I suggest that we merge the PR and if there would be any additional feedback, I will apply it in a separate PR.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**